client-js/client/transport.ts

```diff
@@ -39,29 +22,69 @@ export type Tracks = {
   };
 };
 
+export type TransportConnectionParams = unknown;
+
 export abstract class Transport {
-  protected declare _options: RTVIClientOptions;
+  protected declare _options: PipecatClientOptions;
   protected declare _onMessage: (ev: RTVIMessage) => void;
   protected declare _callbacks: RTVIEventCallbacks;
+  protected declare _abortController: AbortController | undefined;
   protected _state: TransportState = "disconnected";
-  protected _expiry?: number = undefined;
 
   constructor() {}
 
+  /** called from PipecatClient constructor to wire up callbacks */
   abstract initialize(
-    options: RTVIClientOptions,
+    options: PipecatClientOptions,
     messageHandler: (ev: RTVIMessage) => void
   ): void;
 
+  /**
+   * This method is intended to initialize cam/mic devices. It is wrapped
+   * by PipecatClient.initDevices and should not be called directly. It is also
+   * called as part of PipecatClient.connect if it has not already called.
+   */
   abstract initDevices(): Promise<void>;
 
-  abstract connect(
-    authBundle: unknown,
-    abortController: AbortController
-  ): Promise<void>;
-  abstract disconnect(): Promise<void>;
+  connect(connectParams?: TransportConnectionParams): Promise<void> {
+    this._abortController = new AbortController();
+    let validatedParams = connectParams;
+    try {
+      validatedParams = this._validateConnectionParams(connectParams);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (e: any) {
+      throw new RTVIError(
+        `Invalid connection params: ${e.message}. Please check your connection params and try again.`
+      );
+    }
+    return this._connect(validatedParams);
+  }
+
+  abstract _validateConnectionParams(connectParams?: unknown): unknown;
+
+  /**
+   * Establishes a connection with the remote server. This is the main entry
+   * point for the transport to start sending and receiving media and messages.
+   * This is called from PipecatClient.connect() and should not be called directly.
+   * @param abortController
+   */
+  abstract _connect(connectParams?: TransportConnectionParams): Promise<void>;
+  /**
+   * Disconnects the transport from the remote server. This is called from
+   * PipecatClient.disconnect() and should not be called directly.
+   */
```

Comment:
The other comments above are on the abstract methods. Should this move to `abstract _disconnect()`, too?